### DISCUSSION TITLE
Using knex event in adonisjs

### DISCUSSION THREAD

**Discussion by JaconsMorais** [adonisjs/core#3040] [Help] [answered]
Hello guys, I need to use knex event to log all queries made by Lucid for debugging purposes
here the link https://knexjs.org/#Interfaces-query

How can I make this work in adonisjs?

**Comment by varun-nambiar** [comment]
Is this what you are looking for?
https://docs.adonisjs.com/guides/deployment#debugging-database-queries

**Comment by JaconsMorais** [comment, marked as answer]
oh yeah, actually I'm still using v4.1, so the approach is a little different.

This is how I achieve this in v4.1:
```
const db = use('Database')
  const Env = use('Env')

  const shouldLogSQLStatements = Env.get('LOG_SQL', false)

  if (shouldLogSQLStatements)
    db.on('query', (q) => {
      if (q.method in logMethods) {
        console.log(toSQL(q))
      }
    })
```
